### ISSUE TITLE
For symmetry, add functions array:scan-left and array:scan-right

### ISSUE THREAD

**Issue by michaelhkay** [qt4cg/qtspecs#1307]
The fold-left and fold-right functions are defined both for sequences and arrays; symmetry demands that the same should apply to scan-left and scan-right.

**Comment by ChristianGruen** [comment]
Yes, we should do so. My favorite solution would be to offer the array functions for structured operations and the sequence functions for sequence operations, similar to  what does `array:for-each`  and `fn:for-each` (when you use the sequence functions, you can always create structured results if you need to).

**Comment by ChristianGruen** [comment]
I propose to close this. Maybe there is no need for the equivalent functions. The use cases are limited, and the sequence variants create structured results (in contrast to e.g. `fn:for-each`).
